compiler/infra/ForwardList.hpp

```diff
@@ -0,0 +1,684 @@
+#ifndef FORWARD_LIST_HPP
+#define FORWARD_LIST_HPP
+
+#pragma once
+
+#if 0
```

Comment:
At the very least there should be some Doxygen-style header on this file that describes what this data structure is and what its intended use is over and above the C++ std forward_list.